azure/client.go

```diff
@@ -0,0 +1,85 @@
+package azure
+
+import (
+	"encoding/xml"
+	"fmt"
+)
+
+const (
+	errPublishSettingsConfiguration       = "PublishSettingsFilePath is set. Consequently ManagementCertificatePath and SubscriptionId must not be set."
+	errManagementCertificateConfiguration = "Both ManagementCertificatePath and SubscriptionId should be set, and PublishSettingsFilePath must not be set."
+	errParamNotSpecified                  = "Parameter %s is not specified."
+)
+
+// Config is used to configure the creation of a client
+type Config struct {
+	//ManagementCertificatePath is the path to a file containing the management
+	//certificate for the subscription on which this client will operate. If using
+	//this method of authentication, SubscriptionId must also be set, and
+	//PublishSettingsFilePath must not be set.
+	ManagementCertificatePath string
+
+	//SubscriptionId is the name of the subscription on which the subscription will
+	//operate. If using this method of authentication, ManagementCertificatePath must
+	//also be set, and PublishSettingsFilePath must not be set.
+	SubscriptionId string
+
+	//PublishSettingsFilePath is the path to an Azure PublishSettings file containing
+	//the management settings for a subscription. If using this method of authentication,
+	//neither ManagementCertificatePath or SubscriptionId may be set.
+	PublishSettingsFilePath string
+
+	//Anonymous flags whether any authentication options will be specified for the
+	//client. If Anonymous is set to true, this will override any other authentication
+	//options specified.
+	Anonymous bool
```

Comment:
That would be reasonable. Originally there were two different functions for setting the global state.